BrainPortal/app/controllers/demands_controller.rb

```diff
@@ -0,0 +1,437 @@
+      redirect_to '/login'
+      return
+    end
+
+    @demand.destroy
+    flash[:notice] = "The account request has been deleted."
+
+    redirect_to :action => :index
+  end
+
+
+  # Confirms that a demandee's email address actually belongs to them
+  def confirm #:nodoc:
+    @demand = Demand.find(params[:id]) rescue nil
+    token    = params[:token] || ""
+    if @demand.present? && token.present? && @demand.confirm_token == token
+      @demand.confirmed = true
+      @demand.save
+    else
+      redirect_to '/login'
+    end
+  end
+
+
+  # Administrator action that converts a demand into a user
+  def approve
+    @demand = Demand.find(params[:id]) rescue nil
+
+    unless can_edit?(@demand)
+      flash[:error] = "Could not approve account"
+      redirect_to '/login'
+      return
+    end
+
+    if @demand.login.blank?
+      flash[:error] = "Before approval, a 'login' name must be set."
+      redirect_to :action => :edit, :id => @demand.id
+      return
+    end
+
+    symbolic_result, @info, @exception_trace = approve_one(@demand)
+
+    if symbolic_result == :all_ok
+      # flash.now[:notice] = "The account request for '#{@demand.full}' has been approved."
+      # flash.now[:notice] += "\nThe user was notified of his or her new account."
+      current_user.addlog_context(self,"Approved account request for user '#{@demand.login}'")
+      User.find_by_login(@demand.login).addlog_context(self, "Account created after request approved by '#{current_user.login}'")
+    elsif symbolic_result == :failed_save
+      # flash.now[:error] = @info.presence || ""
+    elsif symbolic_result == :failed_approval
+      # flash.now[:error] = @info.presence || ""
+    elsif symbolic_result == :not_notifiable
+      # flash.now[:error] = @info.presence || ""
+      @current_user.addlog_context(self,"Approved account request for user '#{@demand.login}'")
+      User.find_by_login(@demand.login).addlog_context(self, "Account created after request approved by '#{@current_user.login}'")
+    end
+
+  end
+
+
+
+  def multi_action #:nodoc:
+    if params[:commit] =~ /Approve/
+      return approve_multi
+    end
+
+    if params[:commit] =~ /Fix Login/
+      return fix_login_multi
+    end
+
+    if params[:commit] =~ /Resend/
+      return resend_conf_multi
+    end
+
+    if params[:commit] =~ /Delete/
+      return delete_multi
+    end
+
+    redirect_to :action => :index
+  end
+
+
+
+  def delete_multi #:nodoc:
+    reqids = params[:reqids] || []
+    reqs = Demand.find(reqids)
+
+    count = 0
+    reqs.each do |req|
+      count += 1 if req.destroy
+    end
+
+    flash[:notice] = "Deleted " + view_pluralize(count, "record") + "."
+
+    redirect_to :action => :index
+  end
+
+
+
+  def fix_login_multi #:nodoc:
+    reqids = params[:reqids] || []
+    reqs = Demand.find(reqids)
+
+    @results = reqs.map do |req|
+
+      old   = req.login
+      if old =~ /\A[a-z][a-zA-Z0-9]+\z/ && old.size > 3 && old.size < 40
```

Comment:
Validators can allow blanks; within the Demand object validator say "ok" if the field is blank, otherwise delegate to the user model's "login" validator. Something like (not tested):

```ruby
 NormalUser.validators_on(:login).each do |val|
  val.validate(demand) #might have to invoked on new temp NormalUser instead
end 
# extract errors here
```